pkg/backends/azurekeyvault.go

```diff
@@ -29,59 +38,55 @@ func (a *AzureKeyVault) Login() error {
 
 // GetSecrets gets secrets from Azure Key Vault and returns the formatted data
 // For Azure Key Vault, `kvpath` is the unique name of your vault
+// For Azure use the version here not make really sens as each secret have a different version but let support it
 func (a *AzureKeyVault) GetSecrets(kvpath string, version string, _ map[string]string) (map[string]interface{}, error) {
 	kvpath = fmt.Sprintf("https://%s.vault.azure.net", kvpath)
 
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 
-	data := make(map[string]interface{})
+	verboseOptionalVersion("Azure Key Vault list all secrets from vault %s", version, kvpath)
 
-	utils.VerboseToStdErr("Azure Key Vault listing secrets in vault %v", kvpath)
-	secretList, err := a.Client.GetSecretsComplete(ctx, kvpath, nil)
+	client, err := a.ClientBuilder(kvpath, a.Credential, nil)
```

Comment:
Can the client building happen in the login so that we don’t have to call this twice? Then we can reference the client in either method

Reply:
Or in the “new” function